runtime/compiler/trj9/runtime/HookHelpers.cpp

```diff
@@ -141,15 +141,37 @@ namespace  {
    
    // We need to update the nextMethod and prevMethod pointers of the J9JITExceptionTable that
    // point to the old J9JITExceptionTable to now point to the new (stub) J9JITExceptionTable.
-   inline void updateExceptionTableLinkedList(J9JITExceptionTable *stubMetadata)
+   inline void updateExceptionTableLinkedList(J9VMThread *vmThread, J9JITExceptionTable *oldMetadata, J9JITExceptionTable *newMetadata)
       {
-      J9JITExceptionTable *prev = stubMetadata->prevMethod;
-      J9JITExceptionTable *next = stubMetadata->nextMethod;
+      // newMetaData's prevMethod and nextMethod should be the
+      // same as oldMetaData's due to the memcpy
+      J9JITExceptionTable *prev = oldMetadata->prevMethod;
+      J9JITExceptionTable *next = oldMetadata->nextMethod;
       
+      J9Class *j9class = J9_CLASS_FROM_METHOD(oldMetadata->ramMethod);
```

Comment:
I would move these 2 statement in the `else` part, closer to where they are used.